sdk/synapse/synapse-spark/test/utils/utils.common.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { env } from "@azure/test-utils-recorder";
+import * as assert from "assert";
+
+// Async iterator's polyfill for Node 8
+if (!Symbol || !(Symbol as any).asyncIterator) {
+  (Symbol as any).asyncIterator = Symbol.for("Symbol.asyncIterator");
+}
+
+export function getWorkspaceName(): string {
+  const WorkspaceEnvVarName = "WORKSPACE_NAME";
+  const WorkspaceName: string | undefined = env[WorkspaceEnvVarName];
+
+  if (!WorkspaceName) {
+    throw new Error(`${WorkspaceEnvVarName} environment variable not specified.`);
+  }
+
+  return WorkspaceName;
+}
+
+export function getSparkpoolName(): string {
+  const SparkpoolEnvVarName = "SPARKPOOL_NAME";
+  const Sparkpool: string | undefined = env[SparkpoolEnvVarName];
+
+  if (!Sparkpool) {
+    throw new Error(`${SparkpoolEnvVarName} environment variable not specified.`);
+  }
+
+  return Sparkpool;
+}
+
+export async function assertThrowsAbortError(cb: () => Promise<any>): Promise<void> {
```

Comment:
I'm assuming this helper is because chai's assert.throws doesn't work properly with async functions?

Reply:
To be honest, I am not sure for that. I just refer to this helper from [https://github.com/Azure/azure-sdk-for-js/blob/1f8a171a648dcd53de6ebd4b8e5d420fa9c54abf/sdk/keyvault/keyvault-secrets/test/utils/utils.common.ts#L23](https://github.com/Azure/azure-sdk-for-js/blob/1f8a171a648dcd53de6ebd4b8e5d420fa9c54abf/sdk/keyvault/keyvault-secrets/test/utils/utils.common.ts#L23)

---
In reply to: [521617527](https://github.com/Azure/azure-sdk-for-js/pull/12487#discussion_r521617527) [](ancestors = 521617527)